container-run/pkg/containerrun/containerrun.go

```diff
@@ -269,8 +269,12 @@ func (p *ContainerProcess) Signal(sig os.Signal) error {
 
 // Wait waits for the process.
 func (p *ContainerProcess) Wait() error {
-	if _, err := p.process.Wait(); err != nil {
-		return fmt.Errorf("failed to wait for process: %v", err)
+	state, err := p.process.Wait()
+	if err != nil {
+		return fmt.Errorf("failed to run process: %v", err)
+	} else if !state.Success() {
+		err := &exec.ExitError{ProcessState: state}
+		return fmt.Errorf("failed to run process: %v", err)
```

Comment:
It might be nice to have different error messages between "could not run" and "ran, but the process returned an error".

Reply:
It is handled by the error being propagated.